### ISSUE TITLE
Style/TernaryParentheses breaks pattern matching

### ISSUE THREAD

**Issue by jdashton** [rubocop/rubocop#10336]
When Rubocop detects Style/TernaryParentheses, it fails to allow for the low precedence of the `in` keyword with new Ruby 3 pattern matching. 

```
@board[2][room_spot] = (md[1].to_sym in AMPHIPODS) ? md[1].to_sym : :empty
```
 is not the same thing as
```
@board[2][room_spot] = md[1].to_sym in AMPHIPODS ? md[1].to_sym : :empty
```

--------

## Expected behavior

Rubocop should either allow the parens to remain as they are, or offer an even better way of wording this expression.

## Actual behavior

Rubocop removed the parens. This introduces a syntax error that causes Rubocop to report many false positives, such as `Lint/Syntax: module definition in method body`, `Lint/Syntax: class definition in method body`, `Lint/Syntax: unexpected token tEH` and several more. I have elided these in the example below to draw attention to the primary problem.

## Steps to reproduce the problem

Input:
```
          @board[2][room_spot] = (md[1].to_sym in AMPHIPODS) ? md[1].to_sym : :empty
```

Execution:
```
$ bin/rubocop --debug -A lib/aoc2021/puzzles/amphipod.rb
For /Users/jdashton/cSens/learn/aoc2021-ruby: configuration from /Users/jdashton/cSens/learn/aoc2021-ruby/.rubocop.yml
Default configuration from /Users/jdashton/.rvm/gems/ruby-3.1.0@aoc2021/gems/rubocop-1.24.0/config/default.yml
Inspecting 1 file
Scanning /Users/jdashton/cSens/learn/aoc2021-ruby/lib/aoc2021/puzzles/amphipod.rb
E

Offenses:

...other secondary or unrelated offenses elided...

lib/aoc2021/puzzles/amphipod.rb:211:34: C: [Corrected] Style/TernaryParentheses: Omit parentheses for ternary conditions.
          @board[2][room_spot] = (md[1].to_sym in AMPHIPODS) ? md[1].to_sym : :empty
                                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

1 file inspected, 13 offenses detected, 3 offenses corrected
Finished in 0.5465299999341369 seconds
```

Result:
```
          @board[2][room_spot] = md[1].to_sym in AMPHIPODS ? md[1].to_sym : :empty
```

## RuboCop version

```
$ bin/rubocop -V
1.24.0 (using Parser 3.0.3.2, rubocop-ast 1.15.1, running on ruby 3.1.0 x86_64-darwin21)
```
